test/unit/lookup_keys/puppetclass_lookup_key.rb

```diff
@@ -0,0 +1,39 @@
+require 'test_helper'
+
+class PuppetclassLookupKeyTest < ActiveSupport::TestCase
+  test "should not update default value unless override is true" do
```

Comment:
These tests contain a lot of cruft that is unnecessary to test `check_override_selected`. I changed them to not create extra objects but still exercise `check_override_selected` - let me know if you can change them to this:

```ruby
test "should not update default value unless override is true" do
    lookup_key = FactoryGirl.create(:puppetclass_lookup_key,
                                   :key_type => 'string',
                                   :default_value => "test123")
    refute lookup_key.override                                                                                                                  lookup_key.default_value = '33333'                                                                                                          refute lookup_key.valid?
  end

  test "should update description when override is false" do
    lookup_key = FactoryGirl.create(:puppetclass_lookup_key,
                                   :key_type => 'string',
                                   :default_value => "test123",
                                   :description => 'description')
    refute lookup_key.override
    lookup_key.description = 'new_description'
    assert lookup_key.valid?
  end

  test "should allow to uncheck override" do
    lookup_key = FactoryGirl.create(:puppetclass_lookup_key,
                                   :key_type => 'string',
                                   :default_value => "test123",
                                   :override => true)
    lookup_key.override = false
    assert lookup_key.valid?
  end
```